run/jenkins.py

```diff
@@ -0,0 +1,282 @@
+        print('Stopping here (pass PROD_RUN env var to upload results).')
+        return
+
+    print('==================================================')
+    print('Uploading results to gs://%s' % GS_RESULTS_BUCKET)
+
+    # TODO: change this from rsync to cp
+    command = ['gsutil', '-m', '-h', 'Content-Encoding:gzip',
+               'rsync', '-r', args['SHA'], 'gs://wptd/%s' % args['SHA']]
+    return_code = subprocess.check_call(command, cwd=args['output_path'])
+    assert return_code == 0
+    print('Successfully uploaded!')
+    print('HTTP summary URL: %s' % GS_HTTP_RESULTS_URL)
+
+    print('==================================================')
+    print('Creating new TestRun in the dashboard...')
+    url = '%s/test-runs' % PROD_HOST
+
+    if args['prod_run']:
+        final_browser_name = platform['browser_name']
+    else:
+        # The PROD_WET_RUN is identical to PROD_RUN, however the
+        # browser name it creates will be prefixed by eval-,
+        # causing it to not show up in the dashboard.
+        final_browser_name = 'eval-%s' % platform['browser_name']
+
+    response = requests.post(url, params={
+            'secret': args['upload_secret']
+        },
+        data=json.dumps({
+            'browser_name': final_browser_name,
+            'browser_version': platform['browser_version'],
+            'os_name': platform['os_name'],
+            'os_version': platform['os_version'],
+            'revision': args['SHA'],
+            'results_url': GS_HTTP_RESULTS_URL
+        }
+    ))
+    if response.status_code == 201:
+        print('Run created!')
+    else:
+        print('There was an issue creating the TestRun.')
+
+    print('Response status code:', response.status_code)
+    print('Response text:', response.text)
+
+
+def patch_wpt(wptd_path, wpt_path, platform):
+    """Applies util/wpt.patch to WPT.
+
+    The patch is necessary to keep WPT running on long runs.
+    jeffcarp has a PR out with this patch:
+    https://github.com/w3c/web-platform-tests/pull/5774
+    """
+    with open('%s/util/wpt.patch' % wptd_path) as f:
+        patch = f.read()
+
+    # The --sauce-platform command line arg doesn't
+    # accept spaces, but Sauce requires them in the platform name.
+    # https://github.com/w3c/web-platform-tests/issues/6852
+    patch = patch.replace('__platform_hack__', '%s %s' % (
+        platform['os_name'], platform['os_version'])
+    )
+
+    p = subprocess.Popen(
+        ['git', 'apply', '-'], cwd=wpt_path, stdin=subprocess.PIPE
+    )
+    p.communicate(input=patch)
+
+
+def get_and_validate_platform(wptd_path):
+    with open('%s/browsers.json' % wptd_path) as f:
+        browsers = json.load(f)
+
+    platform_id = os.environ['PLATFORM_ID']
+    assert platform_id, 'PLATFORM_ID env var required'
+    assert platform_id in browsers, 'PLATFORM_ID not found in browsers.json'
+    return platform_id, browsers.get(platform_id)
+
+
+def report_to_summary(wpt_report):
+    test_files = {}
```

Comment:
Should there be code sharing between this file and `run.py`?

Reply:
(see above - this file is meant to replace run.py)